config-xml/rule-schema.xsd

```diff
@@ -0,0 +1,303 @@
+	</xs:complexType>
+
+	<xs:complexType name="when-project">
+		<!-- Conditions -->
+		<xs:sequence>
+			<xs:element ref="project" />
+		</xs:sequence>
+	</xs:complexType>
+
+
+	<xs:complexType name="iteration-all">
+		<xs:sequence>
+			<xs:element name="when" type="when-all" />
+			<xs:element name="iteration" type="iteration-all" />
+			<xs:element ref="classification" />
+			<xs:element ref="hint" />
+			<xs:element ref="xslt" />
+			<xs:element ref="lineitem" />
+		</xs:sequence>
+	</xs:complexType>
+
+	<xs:complexType name="iteration-xml">
+		<xs:sequence>
+			<xs:element name="when" type="when-xml" minOccurs="0"
+				maxOccurs="1" />
+			<xs:element name="iteration" type="iteration-all"
+				minOccurs="0" />
+			<xs:element ref="classification" minOccurs="0" />
+			<xs:element ref="hint" minOccurs="0" />
+			<xs:element ref="xslt" minOccurs="0" />
+			<xs:element ref="lineitem" />
+		</xs:sequence>
+	</xs:complexType>
+
+	<xs:complexType name="iteration-project">
+		<xs:sequence>
+			<xs:element name="when-all" type="when-all" />
+			<xs:element name="iteration" type="iteration-all" />
+			<xs:element ref="lineitem" />
+		</xs:sequence>
+	</xs:complexType>
+
+
+	<xs:complexType name="iteration-java">
+		<xs:sequence>
+			<xs:element name="when" type="when-all" />
+			<xs:element name="iteration" type="iteration-all" />
+			<xs:element ref="classification" />
+			<xs:element ref="hint" />
+		</xs:sequence>
+	</xs:complexType>
+
+	<xs:complexType name="rule-java">
+		<xs:sequence>
+			<xs:element name="when" type="when-java">
+			</xs:element>
+			<xs:element name="perform">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="iteration" type="iteration-java">
+						</xs:element>
+					</xs:sequence>
+				</xs:complexType>
+			</xs:element>
+		</xs:sequence>
+	</xs:complexType>
+	
+	<xs:complexType name="rule-xml">
+		<xs:sequence>
+			<xs:element name="when" type="when-xml">
+			</xs:element>
+			<xs:element name="perform">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="iteration" type="iteration-xml">
+						</xs:element>
+					</xs:sequence>
+				</xs:complexType>
+			</xs:element>
+		</xs:sequence>
+	</xs:complexType>
+	
+	<xs:complexType name="rule-project">
+		<xs:sequence>
+			<xs:element name="when" type="when-project">
+			</xs:element>
+			<xs:element name="perform">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="iteration" type="iteration-project">
+						</xs:element>
+					</xs:sequence>
+				</xs:complexType>
+			</xs:element>
+		</xs:sequence>
+	</xs:complexType>
+
+	<xs:complexType name="rule-all">
+		<xs:sequence>
+			<xs:element name="when" type="when-all">
+			</xs:element>
+			<xs:element name="perform">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="iteration" type="iteration-all">
+						</xs:element>
+					</xs:sequence>
+				</xs:complexType>
+			</xs:element>
+		</xs:sequence>
+	</xs:complexType>
+
+
+	<xs:element name="ruleset">
```

Comment:
This is the main core of XSD referencing other "element" and "type" definitions listed above. Some of them are not used yet (rules-xml, rules-java, rules-project with dependencies)